src/renderers/shared/fiber/ReactDebugCurrentFiber.js

```diff
@@ -33,9 +33,13 @@ function getCurrentFiberOwnerName() : string | null {
     }
     switch (fiber.tag) {
       case IndeterminateComponent:
-      case FunctionalComponent:
       case ClassComponent:
         return getComponentName(fiber);
+      case FunctionalComponent:
```

Comment:
If I put `FunctionalComponent` directly with `HostComponent` where it returns `null` when there is no `fiber._debugOwner` then three following tests move from the fiber success to fiber failure:

```
src/renderers/shared/fiber/__tests__/ReactIncrementalErrorHandling-test.js
-* catches reconciler errors in a boundary during mounting
-* catches reconciler errors in a boundary during update

src/renderers/shared/shared/__tests__/ReactComponent-test.js
-* includes owner name in the error about badly-typed elements
```

All three of those tests specifically; include a `FunctionalComponent` returning `const X = undefined; return <X />`.